lib/state/init.js

```diff
@@ -1,16 +1,22 @@
 'use strict';
 const _ = require('lodash');
 const path = require('path');
-const globalModulesPath = require('global-modules');
 const readPackageJson = require('../in/read-package-json');
 const globalPackages = require('../in/get-installed-packages');
 const emoji = require('../out/emoji');
+var globalModulesPath = require('global-modules');
+const fs = require('fs');
 
 function init(currentState, userOptions) {
     return new Promise((resolve, reject) => {
         _.each(userOptions, (value, key) => currentState.set(key, value));
 
         if (currentState.get('global')) {
+
+            if (process.env.NODE_PATH && fs.existsSync(process.env.NODE_PATH)) {
+                globalModulesPath = process.env.NODE_PATH;
```

Comment:
I would prefer it if we didn't modify this variable, how about something like this instead:

```js
currentState.set('cwd', process.env.NODE_PATH || globalModulesPath)
```

or

```js
const modulesPath = process.env.NODE_PATH || globalModulesPath
```

since it's used multiple times...

I'm also not sure that we should check `fs.existsSync(process.env.NODE_PATH)`, if someone has set `NODE_PATH` but has a small typo, it would be better to exit with an error instead of silently not using it.

What we could possibly do is to error out early if the directory doesn't exist. That should be done both when `NODE_PATH` is provided, and when it isn't. It should probably also happen for non-global runs as well, so that check should be placed around line 25 (old)/31 (new), after the if branches.

Something like this:

```js
if (currentState.get('global')) {
  const modulesPath = process.env.NODE_PATH || globalModulesPath

  currentState.set('cwd', modulesPath);
  currentState.set('nodeModulesPath', modulesPath);
  currentState.set('globalPackages', globalPackages(modulesPath));
} else {
  // ...
}

// potentially check fs.exists or fs.access on currentState.get `cwd`, `nodeModulesPath`, etc.
```